base/src/com/google/idea/blaze/base/sync/projectstructure/ModuleEditorImpl.java

```diff
@@ -119,8 +120,8 @@ public void commitWithGc(BlazeContext context) {
     List<Module> orphanModules = Lists.newArrayList();
     for (Module module : ModuleManager.getInstance(project).getModules()) {
       if (!modules.containsKey(module.getName())
-              && OrphanedModuleValidator.EP_NAME.getExtensionList().stream().anyMatch(ep -> ep.isOrphanModule(module))) {
-        orphanModules.add(module);
+              && (ModuleValidator.EP_NAME.getExtensionList().isEmpty() || ModuleValidator.EP_NAME.getExtensionList().stream().anyMatch(n -> !n.isModuleValid(module)))) {
```

Comment:
This is a wrong condition... if at least ANY "module consumer" is interested it cannot be `orphaned`, with your current logic if any consumer is not interested module will be `orphaned`